### ISSUE TITLE
esFtl Size Limit ??

### ISSUE THREAD

**Issue by ghost** [thearistotlemethod/esftl#1]
The README.md implies there is a size limitation.  Is that true ??
I’m going to try and use this with a 4Gb Micron SLC NAND part.  (Upgrading to the 8Gb at some point.)

Thanks

**Comment by thearistotlemethod** [comment]
Yes, there is a limit due to usage of 16 bit integer on mapping between logical sectors to nand pages. Current code base can only address 65535 pages and that means 128 mb if the page size is 2kb. However, I think it can be overcome by making some changes in light of this comment. 

Btw, this project aims to use small memory as much as possible. It is the main reason of being used 16 bit instead of 32 bit.

**Comment by ghost** [comment]
The Micron MT29F4G01ABAFD12 part we're using has a 4K page size and I think 131072 total pages.

**Comment by thearistotlemethod** [comment]
MT29F4G01ABAFD12
– Page size ×1: 4352 bytes (4096 + 256 bytes)
– Block size: 64 pages (256K + 16K bytes)
– Plane size: 1 × 2048 blocks

It seems it has 131072 pages, but the library is capable of addressing only 65535 pages. That means you can only use half.
